src/main/java/org/mule/extension/http/internal/listener/HttpRequestToResult.java

```diff
@@ -104,6 +107,10 @@
     return Result.<Object, HttpRequestAttributes>builder().output(payload).mediaType(mediaType).attributes(attributes).build();
   }
 
+  private static MediaType getJavaMediaType(MediaType mediaType) {
+    return MediaType.create(APPLICATION_JAVA.getPrimaryType(), APPLICATION_JAVA.getSubType(), mediaType.getCharset().get());
```

Comment:
This can be replaced with `APPLICATION_JAVA.charset(mediaType.getCharset().get()`